tsdb/engine/tsm1/engine.go

```diff
@@ -2096,11 +2118,39 @@ func (e *Engine) ShouldCompactCache(t time.Time) bool {
 	return t.Sub(e.Cache.LastWriteTime()) > e.CacheFlushWriteColdDuration
 }
 
+// isFileOptimized returns true if a TSM file appears to have already been previously optimized.
+// If file appears previously optimized, a description of the heuristic used to determine this is also returned.
+func (e *Engine) isFileOptimized(f string) (bool, string) {
+	if tsmPointsPerBlock := e.Compactor.FileStore.BlockCount(f, 1); tsmPointsPerBlock >= e.CompactionPlan.GetAggressiveCompactionPointsPerBlock() {
+		return true, fmt.Sprintf("first block of file contains aggressive points per block (%d >= %d)", tsmPointsPerBlock, e.CompactionPlan.GetAggressiveCompactionPointsPerBlock())
+	} else {
+		return false, ""
+	}
+}
+
+// IsGroupOptimized returns true if any file in a compaction group appears to be have been previously optimized.
+// The name of the first optimized file found along with the heuristic used to determine this is returned.
+func (e *Engine) IsGroupOptimized(group CompactionGroup) (optimized bool, file string, heuristic string) {
+	for _, f := range group {
+		if isOpt, heur := e.isFileOptimized(f); isOpt {
+			return true, f, heur
+		}
+	}
+	return false, "", ""
+}
+
+const waitForOptimization = time.Hour
+const tickPeriod = time.Second
+
+var ticksBeforeOptimize = int(waitForOptimization.Seconds())
```

Comment:
`ticksBeforeOptimize` is only correct if `tickPeriod` is `time.Second`.

I think this will generalize it if `tickPeriod` is not `time.Second`:
```go
var ticksBeforeOptimize = int(waitForOptimization.Seconds() * time.Second / tickPeriod)
```